ucsb-cs156-f24/proj-happycows-f24-10
# added web integration tests for buying and selling cows

## Overview

In this PR, I created web integration tests for buying and selling cows, checking that the total wealth, total cows bought, and total cows sold parameters correctly update. I bought 42 cows and sold 10, so wealth was checked to be 5800 after the first buy, and after selling it was checked to be 6800, and of course the total cows bought to be 42 and total cows sold to be 10. This does not really affect the user. 

## Feedback Request 

Please give suggestions on the following:
- There could be more parameters to check. Any suggestions to more thoroughly test are welcome. 

## Future Possibilities 
- In general, we could add more web integration testing for other features. This just tested a few, but there are plenty more to check. 

## Tests 
1. Download this branch.
2. Run the following commands in backend:
- mvn clean
- INTEGRATION=true mvn test-compile
- INTEGRATION=true HEADLESS=false mvn test-compile failsafe:integration-test -Dit.test=UserCommonsWebIT.java
3. There should be no errors when running the test

## Linked Issues

Closes #62
